pkg/common/tls/tls.go

```diff
@@ -0,0 +1,54 @@
+package tls
+
+import (
+	ctls "crypto/tls"
+	"fmt"
+	"os"
+)
+
+type TLSVersion string
+
+const (
+	TLS10 TLSVersion = "TLS10"
+	TLS11 TLSVersion = "TLS11"
+	TLS12 TLSVersion = "TLS12"
+	TLS13 TLSVersion = "TLS13"
+)
+
+type tlsEnvVariableName string
+
+const (
+	minHttpTlsVersionEnv tlsEnvVariableName = "KEDA_HTTP_MIN_TLS_VERSION"
+	minGrpcTlsVersionEnv tlsEnvVariableName = "KEDA_GRPC_MIN_TLS_VERSION"
+)
+
+const (
+	defaultMinHttpTlsVersion = TLS12
+	defaultMinGrpcTlsVersion = TLS13
+)
+
+func getMinTlsVersion(envKey tlsEnvVariableName, defaultVal TLSVersion) (uint16, error) {
+	version := string(defaultVal)
+	if val, ok := os.LookupEnv(string(envKey)); ok {
+		version = val
+	}
+	mapping := map[string]uint16{
+		string(TLS10): ctls.VersionTLS10,
+		string(TLS11): ctls.VersionTLS11,
```

Comment:
not sure if it's a good idea to enable TLS 1.0 and 1.1, these are not considered secure anymore

Reply:
yes, probably makes more sense

Reply:
Even though this will make this change a breaking change for those brave people who still wants to configure it with older TLS... 🫨 
should we do it in a different PR?

Reply:
I defer to @kedacore/keda-core-contributors for their opinion, maybe we announce deprecation first and then remove TLS 1.0 and 1.1 in KEDA 2.18 or 2.19?

Reply:
Yeah

Reply:
I can probably add  a warning print  in case TLS10 or TLS11 was requested:
```go
if version == string(TLS11) || version == string(TLS10) {
		ctrl.Log.WithName("tls_setup").Info(fmt.Sprintf("Warning: Min TLS value %s will be deprecated in Keda 2.18",  version))
	}
```

Reply:
I think that we should allow TLS10 until it's deprecated/removed, despite the security of it. We are using TLS1.2 as minimum by default, so I'd just keep them as optional for the users if they enable them